tools/ampir/ampir.xml

```diff
@@ -0,0 +1,85 @@
+<tool name="ampir" id="ampir" version="0.01">
+  <!--Source in git at: https://github.com/fubar2/galaxy_tf_overlay-->
+  <!--Created by toolfactory@galaxy.org at 01/09/2023 20:56:07 using the Galaxy Tool Factory.-->
+  <description>A toolkit to predict antimicrobial peptides from protein sequences on a genome-wide scale.</description>
+  <requirements>
+    <requirement version="1.1.0" type="package">r-ampir</requirement>
+  </requirements>
+  <stdio>
+    <exit_code range="1:" level="fatal"/>
+  </stdio>
+  <version_command><![CDATA[echo "0.01"]]></version_command>
+  <command><![CDATA[#set prefix = "ampir"
+#set task_process = "ampir"
+Rscript $runme]]></command>
+  <configfiles>
+    <configfile name="runme"><![CDATA[#set prefix = "ampir"
+#set task_process = "ampir"
+#!/usr/bin/env Rscript
+library(ampir)
+input_seqs = read_faa('${faa}')
+prediction = predict_amps(input_seqs,${min_length},model = '${model}')
+prediction = prediction[which(prediction\$prob_AMP >= as.numeric(${min_probability})), ]
+output_seqs = input_seqs[row.names(prediction), ]
+write.table(prediction, file = "$amps_tsv", row.names = FALSE, sep = "\t", quote = FALSE, dec = '.')
+df_to_faa(output_seqs, "$amps_faa")
+version_file_path = "versions.yml"
+version_ampir = paste(unlist(packageVersion("ampir")), collapse = ".")
+f = file(version_file_path, "w")
+writeLines('"${task_process}":', f)
+writeLines("    ampir: ", f, sep = "")
+writeLines(version_ampir, f)
+close(f)]]></configfile>
+  </configfiles>
+  <inputs>
+    <param name="faa" type="data" optional="false" label="FASTA file containing amino acid sequences" help="" format="fasta" multiple="false"/>
+    <param name="min_length" type="integer" value="10" label="Minimum protein length for which predictions will be generated" help=""/>
+    <param name="min_probability" type="float" value="0.7" label="Cut-off for AMP prediction" help=""/>
+    <param name="model" type="select" label="Built-in model for AMP prediction" help="">
+      <option value="precursor">precursor</option>
+      <option value="mature">mature</option>
+    </param>
+  </inputs>
+  <outputs>
+    <data name="amps_faa" format="fasta" label="File containing AMP predictions in amino acid FASTA format" hidden="false"/>
+    <data name="amps_tsv" format="tsv" label="File containing AMP predictions in TSV format" hidden="false"/>
+  </outputs>
+  <tests>
+    <test>
+      <output name="amps_faa" value="amps_faa_sample" compare="sim_size" delta="1000"/>
+      <output name="amps_tsv" value="amps_tsv_sample" compare="sim_size" delta="1000"/>
+      <param name="faa" value="faa_sample"/>
+      <param name="min_length" value="10"/>
+      <param name="min_probability" value="0.7"/>
+      <param name="model" value="precursor"/>
+    </test>
+  </tests>
+  <help><![CDATA[description: A toolkit to predict antimicrobial peptides from protein sequences on a genome-wide scale.
+
```

Comment:
Currently, it is grabbed from the yaml - can't really see any other useful sources but happy to use anything that you'd like to see.
**_This should all be automated._** Manual fixes are not going to be part of these tools since the whole point is that they can be re-autogenerated whenever the underlying nf-core repository is updated in CI?